### ISSUE TITLE
Upgrading to .NET 9 cause build time too long

### ISSUE THREAD

**Issue by SaleSwift** [dotnet/maui#25900]
### Description

In .NET 8 build takes around 5 minutes After i upgrade to .NET 9 it takes around 20-25 minutes

### Version with bug

9.0.10 SR1

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

9.0.10 SR1

### Affected platforms

Windows

### Relevant log output

```shell

```

**Comment by jfversluis** [comment]
Can you give some more details? How do you trigger a build? What is the command you use with all the parameters? Would you be able to provide a [binlog](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs)? Ideally from the old build which was fast and the new build that is slow?

Also see: https://github.com/dotnet/maui/issues/25022

**Comment by SaleSwift** [comment]
@jfversluis I think the log file cant be uploaded because its too large file ,
can i share you screen or give you controll so you check what ever you want ?

**Comment by SaleSwift** [comment]
Additionally, I’ve noticed that when I build the solution twice, it doesn't recognize that the solution is up-to-date on the second build. Instead of saying "up-to-date", it starts rebuilding everything again. This is something that has become noticeable after upgrading to .NET 9. It seems like incremental builds aren't working properly, which adds unnecessary time to the build process.

**Comment by maonaoda** [comment]
For me, after the 9.0GA release, debug mode was spending too much time on AOTCompile.
When using rc2, the build time does not seem to be very long.

**Comment by SaleSwift** [comment]
> For me, after the 9.0GA release, debug mode was spending too much time on AOTCompile. When using rc2, the build time does not seem to be very long.

I see at the build output that take time is the 
Accepted compressed asset
Processing compressed asset
is those releated to AOTCompile 
@maonaoda

**Comment by SaleSwift** [comment]
Any solution for this issue?

**Comment by matteomonizza** [comment]
Same here, very slow build, for iOS projects

**Comment by KaeTar** [comment]
Same here..  debug build happens in seconds but release builds or publishing is well see for yourself:
![Image](https://github.com/user-attachments/assets/e333d1b0-10e8-4fdc-9019-8ef887b0a00f)
this one took 55 minutes.  54 of which was in the AOT compile stage.  Has anyone found a workaround?

**Comment by bcaceiro** [comment]
I am getting these ridiculous times in release mode. Any advance on this subject? It is not feasible

**Comment by SamGustinHdr** [comment]
Same problem here using github actions to build a maui library in release mode. I am using this command to trigger the build: dotnet build --no-restore --configuration Release

**Comment by KaeTar** [comment]
I turned off llvm on the release builds and I am back down to 2 min builds.

`/p:MtouchUseLlvm=false`

`<MtouchUseLlvm>false</MtouchUseLlvm>`

**Comment by AndriyGlodan** [comment]
![Image](https://github.com/user-attachments/assets/9dd60990-bcad-4558-a197-bb7a84b69542)

It took me 18 minutes to build an APK file in Debug mode. How can I disable building APK In Debug mode?

I tried this configuration, but it didn't help

```
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
        <SkipApkEmbed>true</SkipApkEmbed>
    </PropertyGroup>

    <!-- Override the _BuildApkEmbed target when SkipApkEmbed is true -->
    <Target Name="_BuildApkEmbed" Condition="'$(SkipApkEmbed)'=='true'" DependsOnTargets="">
        <Message Text="Skipping _BuildApkEmbed in Debug mode." Importance="high" />
    </Target>
```

**Comment by jeanplevesque** [comment]
Same here. Simply be upgrading from net8.0-ios to net9.0-ios, our build times doubled.

**Comment by Ducvu1996** [comment]
Any solution for this issue?

**Comment by MLB-BIOMETRIC** [comment]
I also have issues! Its taking forever to build... I just upgraded from .NET 8 to .NET 9

![Image](https://github.com/user-attachments/assets/c1382dc5-527f-456d-bdf3-4b1c69183ca0)

**Comment by KaeTar** [comment]
We are still having this issue as well.   And with azure devops having limited apple resources our builds keep hitting the 60 minute max time that azure allows a build job to run and get cancelled.  As a result I am having to manually build and deploy which takes right at 30 min if I do the build on an M4 mac.   I am sure even Azure would like a solution to this as it is just burning cpu time only to get cancelled anyway.

**Comment by am-kh** [comment]
Not planned?
The iOS build is very slow; it takes 2 hours since the migration to .NET 9.

**Comment by matteomonizza** [comment]
What does it mean Not Planned ????????? This is not a feature...!!

**Comment by jadenrogers** [comment]
I had a similar issue for release builds, but was made worse by publishing symbols to Google with "FirebaseCrashlyticsUploadSymbolsEnabled". 

Using the ```<MtouchLink>SdkOnly</MtouchLink>``` brought my builds back to pre .net 9 times, maybe it will help you too.

**Comment by jeremy-visionaid** [comment]
The original bug here seems to reference Windows builds. For Apple targets there is a runtime issue with some movement expected next Patch Tuesday:
https://github.com/dotnet/runtime/issues/110406

I've also posted a partial workaround for debug builds on that thread.

**Comment by LeoJHarris** [comment]
> What does it mean Not Planned ????????? This is not a feature...!!

@matteomonizza I genuinely cracked up at that because its getting comedic, but on a more serious note I think `EnableLLVM` on our Release Android build (maybe among others too) is causing a ridiculous build time to the point I don't even know if VS has got stuck. I'm not honestly not sure if we forfeit these benefits in favor of actually being able to build the project 🤦‍♂

**Comment by RezaRahmatzadeh** [comment]
I'm experiencing extremely poor performance when adding or applying migrations in .NET 9. While a normal build completes in a reasonable time, any EF Core migration operation completely freezes my high-performance laptop and takes around 5 minutes just to generate a small, simple migration. This has been seriously affecting my productivity. I hope this issue gets some attention soon.

**Comment by mike-normal13** [comment]
I spent around half of my workday trying to bypass the AOT compilation by doing this sort of thing in my main .csproj file:

 <PropertyGroup Condition="'$(Configuration)' == 'Debug' and '$(BuildForDevice)' == 'false'">
    <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
    <TargetArchitecture>x64</TargetArchitecture>
    <RunAOTCompilation>false</RunAOTCompilation>
    <UseInterpreter>true</UseInterpreter>
    <MtouchLink>None</MtouchLink>
  </PropertyGroup>

And I tried to run against a x64 Simulator I made.
No love, AOT compiling always happens anyway with these changes.
It would be very nice If I did not have to wait 5 minutes for a local build to run.

.NET 9 is cold wet trash.
